core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -80,8 +85,49 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   //periodically emit metrics (don't need to do this for each message!)
   context.system.scheduler.schedule(30.seconds, 10.seconds, self, EmitMetrics)
 
+  // Key is `kind,memory,time`, value is the number of cold Start in minute
+  val coldStartCount = TrieMap[String, AtomicInteger]()
```

Comment:
this can be a `var immutable.Map` instead, i think?

Reply:
Actually, I would probably not create a separate map here, but rather add `expiration:Option[Deadline]`  to `PreWarmedData` class. Then you can track expiration AND more easily filter to find expired prewarms.

Reply:
Already fixed